components/AppBar/examples/AppBarModalPresentationExample.m

```diff
@@ -0,0 +1,269 @@
+@end
+
+@implementation AppBarModalPresentationExample
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+
+  self.tableView.delegate = self;
+
+  self.appBar.headerViewController.headerView.trackingScrollView = self.tableView;
+  [self.appBar addSubviewsToParent];
+
+  self.tableView.layoutMargins = UIEdgeInsetsZero;
+  self.tableView.separatorInset = UIEdgeInsetsZero;
+
+  self.navigationItem.rightBarButtonItem = [[UIBarButtonItem alloc] initWithTitle:@"Detail"
+                                                                      style:UIBarButtonItemStyleDone
+                                                                              target:self
+                                                                    action:@selector(presentModal)];
+}
+
+- (void)presentModal {
+  AppBarModalPresentationExamplePresented *modalVC =
+      [[AppBarModalPresentationExamplePresented alloc] init];
+  [self presentViewController:modalVC animated:YES completion:^{}];
+}
+
+#pragma mark - UIScrollViewDelegate
+
+- (void)scrollViewDidScroll:(UIScrollView *)scrollView {
+  if (scrollView == self.appBar.headerViewController.headerView.trackingScrollView) {
+    [self.appBar.headerViewController.headerView trackingScrollViewDidScroll];
+  }
+}
+
+- (void)scrollViewDidEndDecelerating:(UIScrollView *)scrollView {
+  if (scrollView == self.appBar.headerViewController.headerView.trackingScrollView) {
+    [self.appBar.headerViewController.headerView trackingScrollViewDidEndDecelerating];
+  }
+}
+
+- (void)scrollViewDidEndDragging:(UIScrollView *)scrollView willDecelerate:(BOOL)decelerate {
+  MDCFlexibleHeaderView *headerView = self.appBar.headerViewController.headerView;
+  if (scrollView == headerView.trackingScrollView) {
+    [headerView trackingScrollViewDidEndDraggingWillDecelerate:decelerate];
+  }
+}
+
+- (void)scrollViewWillEndDragging:(UIScrollView *)scrollView
+                     withVelocity:(CGPoint)velocity
+              targetContentOffset:(inout CGPoint *)targetContentOffset {
+  MDCFlexibleHeaderView *headerView = self.appBar.headerViewController.headerView;
+  if (scrollView == headerView.trackingScrollView) {
+    [headerView trackingScrollViewWillEndDraggingWithVelocity:velocity
+                                          targetContentOffset:targetContentOffset];
+  }
+}
+
+@end
+
+@implementation AppBarModalPresentationExample (CatalogByConvention)
+
++ (NSArray *)catalogBreadcrumbs {
+  return @[ @"App Bar", @"Modal Presentation" ];
+}
+
+- (BOOL)catalogShouldHideNavigation {
+  return YES;
+}
+
+@end
+
+@implementation AppBarModalPresentationExample (TypicalUse)
+
+- (id)init {
+  self = [super init];
+  if (self) {
+    _appBar = [[MDCAppBar alloc] init];
+    _appBar.navigationBar.tintColor = [UIColor whiteColor];
+    _appBar.navigationBar.titleTextAttributes =
+    @{NSForegroundColorAttributeName : [UIColor whiteColor]};
+    [self addChildViewController:_appBar.headerViewController];
+
+    self.title = @"Modal Presentation";
```

Comment:
Because this is a subclass UIKit's UIViewController, we don't have access to backing instance variables to any of the properties associated with it such as title. Agreed in general but we "use of undeclared identifier" error if we try setting _title.